packages/react/toast/src/Toast.tsx

```diff
@@ -195,34 +204,63 @@ const ToastViewport = React.forwardRef<ToastViewportElement, ToastViewportProps>
       }
     }, [context.isClosePausedRef]);
 
+    const getSortedCandidates = React.useCallback(
+      ({ direction }: { direction: 'forwards' | 'backwards' }) => {
+        const items = getItems();
+        const candidateGroups = items.map((item) => {
+          const itemNode = item.ref.current!;
+          const itemGroup = [itemNode, ...getTabbableCandidates(itemNode)];
+          return direction === 'forwards' ? itemGroup.reverse() : itemGroup;
+        });
+        return (direction === 'forwards' ? candidateGroups : candidateGroups.reverse()).flat();
+      },
+      [getItems]
+    );
+
     React.useEffect(() => {
       const viewport = ref.current;
-      // Re-order DOM so most recent toasts are at top of DOM structure to improve tab order
+      // We programmatically manage tabbing as we are unable to influence
+      // the source order with portals, this allows us to reverse the
+      // tab order so that it runs from most recent toast to least
       if (viewport) {
-        const prepended: Set<Node> = new Set();
-        const observer = new MutationObserver((mutations) => {
-          const addedNodes = mutations
-            .map((mutation) => Array.from(mutation.addedNodes))
-            .reduce((a, b) => a.concat(b));
-
-          addedNodes.forEach((node) => {
-            // mutation will immediately fire again when we prepend so we only prepend if
-            // it hasn't just prepended.
-            if (!prepended.has(node)) {
-              viewport.prepend(node);
-              prepended.add(node);
+        const handleKeyDown = (event: KeyboardEvent) => {
+          const isMetaKey = event.altKey || event.ctrlKey || event.metaKey;
+          const isTabKey = event.key === 'Tab' && !isMetaKey;
+
+          if (isTabKey) {
+            const focusedElement = document.activeElement;
+            const isMovingBackwards = event.shiftKey;
+            const targetIsViewport = event.target === viewport;
+
+            // If we're back tabbing after jumping to the viewport then we simply
+            // proxy focus out to the preceding document
+            if (targetIsViewport && isMovingBackwards) {
+              headFocusProxyRef.current?.focus();
+              return;
+            }
+
+            // Otherwise manually reverse the tab order
+            const direction = isMovingBackwards ? 'forwards' : 'backwards';
```

Comment:
This will now read nicer after chaning the param name in `getSortedCandidates`:

```suggestion
            const tabbingDirection = isTabbingBackwards ? 'forwards' : 'backwards';
```